HW3_spark_smythe.py

```diff
@@ -1,156 +0,0 @@
-"""
-  Created by Ed Smythe Sept 2020
-  BDA 696 @ SDSU
-  Current run time is ~ xxxx
-  Baseball ⚾
-  Load this dataset into Spark directly from MariaDB
-  Calculate the batting average using an Transformer
-  Rolling (over last 100 days)
-  Look at the last 100 days that player was in prior to this game
-  Hints
-  You already did it earlier, use that logic to your advantage
-  Make sure the calculation works first
-  Then worry about wrapping it up as a transformer
-  Only get the needed raw data from MariaDB
-  No calculations in MariaDB
-  Do all the logic in Spark
-"""
-
-import sys
-
-from pyspark import StorageLevel, keyword_only
-from pyspark.ml import Pipeline, Transformer
-from pyspark.ml.classification import RandomForestClassifier
-from pyspark.ml.feature import CountVectorizer
-from pyspark.ml.param.shared import HasInputCols, HasOutputCol
-from pyspark.ml.util import DefaultParamsReadable, DefaultParamsWritable
-from pyspark.sql import SparkSession
-from pyspark.sql.functions import col, concat, lit, split, when
-from split_column_transform import SplitColumnTransform
-
-
-# copy from class notes for custom trasnform
-class SplitColumnTransform(
-    Transformer,
-    HasInputCols,
-    HasOutputCol,
-    DefaultParamsReadable,
-    DefaultParamsWritable,
-):
-    @keyword_only
-    def __init__(self, inputCols=None, outputCol=None):
-        super(SplitColumnTransform, self).__init__()
-        kwargs = self._input_kwargs
-        self.setParams(**kwargs)
-        return
-
-    @keyword_only
-    def setParams(self, inputCols=None, outputCol=None):
-        kwargs = self._input_kwargs
-        return self._set(**kwargs)
-
-    def _transform(self, dataset):
-        input_cols = self.getInputCols()
-        output_col = self.getOutputCol()
-
-        # Generate a list of columns to append
-        concat_list = []
-        for count, column in enumerate(input_cols):
-            column_func = when(col(column).isNull(), lit("")).otherwise(col(column))
-            concat_list.append(column_func)
-            if count < len(input_cols) - 1:
-                concat_list.append(lit(" "))
-        dataset = dataset.withColumn(
-            output_col, split(concat(*concat_list), pattern=" ")
-        )
-        return dataset
-
-
-def main():
-    # Setup Spark
-    spark = SparkSession.builder.master("local[*]").getOrCreate()
-
-    # set up sql defaults
-    database = "baseball"
-    port = "3306"
-    user = "root"
-    password = "x11desktop"  # pragma: allowlist secret
-
-    # get table and make batter df
-    bb_df = (
-        spark.read.format("jdbc")
-        .options(
-            url=f"jdbc:mysql://localhost:{port}/{database}",

```

Comment:
considerr using "zeroDateTimeBehavior=convertToNull" to account for null dates

Reply:
I agree with Brent! It would help and you could add it to your URL.

Reply:
ok thanks!